test/e2e/testutils/storage/storage.go

```diff
@@ -0,0 +1,160 @@
+	cl, err := client.New(config, client.Options{})
+	if err != nil {
+		return err
+	}
+
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   namespace,
+			Labels: testutils.GetNamespaceLabel(true),
+		},
+	}
+
+	if err = cl.Create(ctx, ns); err != nil {
+		return err
+	}
+
+	statefullSet := &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      StatefulSetName,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"app": StatefulSetName,
+			},
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Replicas: pointer.Int32(2),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": StatefulSetName,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": StatefulSetName,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "tester",
+							Image: "nginx",
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "liqo-storage-claim",
+									MountPath: "/usr/share/nginx/html",
+								},
+							},
+						},
+					},
+					// put pods in anti-affinity, and prefer local cluster. In this way one pod will be local, the other remote
+					Affinity: &corev1.Affinity{
+						NodeAffinity: &corev1.NodeAffinity{
+							PreferredDuringSchedulingIgnoredDuringExecution: []corev1.PreferredSchedulingTerm{
+								{
+									Weight: 2,
+									Preference: corev1.NodeSelectorTerm{
+										MatchExpressions: []corev1.NodeSelectorRequirement{
+											{
+												Key:      consts.TypeLabel,
+												Operator: corev1.NodeSelectorOpDoesNotExist,
+											},
+										},
+									},
+								},
+							},
+						},
+						PodAntiAffinity: &corev1.PodAntiAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+								{
+									LabelSelector: &metav1.LabelSelector{
+										MatchLabels: map[string]string{
+											"app": StatefulSetName,
+										},
+									},
+									TopologyKey: "kubernetes.io/hostname",
+								},
+							},
+						},
+					},
+				},
+			},
+			VolumeClaimTemplates: []corev1.PersistentVolumeClaim{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "liqo-storage-claim",
+					},
+					Spec: corev1.PersistentVolumeClaimSpec{
+						AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
+								corev1.ResourceStorage: resource.MustParse("1Gi"),
+							},
+						},
+						StorageClassName: pointer.StringPtr("liqo"),
+					},
+				},
+			},
+		},
+	}
+
+	return cl.Create(ctx, statefullSet)
+}
+
+// WaitDemoApp waits until each pod in the StatefulSet is ready.
+func WaitDemoApp(t ginkgo.GinkgoTInterface, options *k8s.KubectlOptions) {
+	var pods []corev1.Pod
+	for len(pods) != 2 {
```

Comment:
https://github.com/gruntwork-io/terratest/blob/v0.38.5/modules/k8s/pod.go#L72 To avoid the sleep until pods are created, what about WaitUntilNumPodsCreatedE?